featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -136,6 +136,42 @@ def day(vals):
         return day
 
 
+class DayOfYear(TransformPrimitive):
+    """Determines the ordinal day of the year from the given datetime
+
+    Description:
+        For a list of dates, return the ordinal day of the year
+        from the given datetime.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> dates = [datetime(2019, 1, 1),
+        ...          datetime(2020, 12, 31),
+        ...          datetime(2020, 2, 28)]
+        >>> dayOfYear = DayOfYear()
+        >>> dayOfYear(dates).tolist()
+        [1, 366, 59]
+    """
+
+    name = "day_of_year"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(
+        logical_type=Ordinal(order=list(range(1, 366))), semantic_tags={"category"}
+    )
+    compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
+    description_template = "the day of year from {}"
+
+    def get_function(self):
+        def dayOfYear(vals):
+            try:
+                return vals.dt.day_of_year
+            except AttributeError:
```

Comment:
Let's skip Spark support for now for this primitive.

Reply:
and dask?

Reply:
I printed the dir of each type's dt attributes. Spark was failiing because it doesn't implement a day_of_year attribute, as it is called dayofyear. However I looked closely at the dir for Pandas and Dask and they support both a day_of_year as well as a dayofyear. In this case I think we can just return dayofyear always, and it would work for all three

Reply:
Okay let's do that